File: java/ElektronikPortable.java
public class ElektronikPortable extends Elektronik {
    private String Durasi_baterai;
    private float Berat;
    private String Tipe_koneksi;

    public ElektronikPortable(int id, String Nama_produk, float Harga, int Stok, String GambarPath, String Kategori_produk, String Deskripsi_produk, String Durasi_baterai, float Berat, String Tipe_koneksi) {
        super(id, Nama_produk, Harga, Stok, GambarPath, Kategori_produk, Deskripsi_produk);
        this.Durasi_baterai = Durasi_baterai;
        this.Berat = Berat;
        this.Tipe_koneksi = Tipe_koneksi;
    }

    @Override
    public void display() {
        super.display();
        System.out.printf(" %-15s %-10.2f %-10s%n", Durasi_baterai, Berat, Tipe_koneksi);
    }
}

File: c++/kode program/ElektronikPortable.cpp
#include "ElektronikPortable.h"
#include <iomanip>

ElektronikPortable::ElektronikPortable(int id, std::string Nama_produk, float Harga, int Stok, std::string GambarPath, std::string Kategori_produk, std::string Deskripsi_produk, std::string Durasi_baterai, float Berat, std::string Tipe_koneksi)
    : Elektronik(id, Nama_produk, Harga, Stok, GambarPath, Kategori_produk, Deskripsi_produk), Durasi_baterai(Durasi_baterai), Berat(Berat), Tipe_koneksi(Tipe_koneksi) {}

void ElektronikPortable::display() const {
    Elektronik::display();
    std::cout << std::left
              << std::setw(15) << Durasi_baterai
              << std::setw(10) << Berat
              << std::setw(10) << Tipe_koneksi
              << std::endl;
}
